### ISSUE TITLE
Improve AutoComplete implementation for Accessibility

### ISSUE THREAD

**Issue by mertsincan** [primefaces/primevue#2822]
MultiSelect's codebase has been completely overhauled and is now closer to spec. In addition, some extra changes were made to the design based on user feedback.
Similar to;
https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-list.html
https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-autocomplete-both.html

**_Added Properties:_**

- [x] **optionLabel**: _Property name or getter function to use as the label of an option._
- [x] **optionDisabled**: _Property name or getter function to use as the disabled flag of an option, defaults to false when not defined._
- [x] **placeholder**: _Default text to display when no option is selected.._
- [x] **disabled**: _When present, it specifies that the component should be disabled._
- [x] **dataKey**: _A property to uniquely identify an option._
- [x] **inputId**: _Identifier of the underlying input element._
- [x] **inputProps**: _Uses to pass all properties of the HTMLInputElement to the focusable input element inside the component_
- [x] **panelStyle**: _Inline style of the overlay panel_
- [x] **panelProps**: _Uses to pass all properties of the HTMLDivElement to the overlay panel inside the component._
- [x] **filterInputProps**: _Uses to pass all properties of the HTMLInputElement to the filter input inside the component._
- [x] **closeButtonProps**: _Uses to pass all properties of the HTMLButtonElement to the clear button inside the overlay panel._
- [x] **autoOptionFocus**: _Whether to focus on the first visible or selected element when the overlay panel is shown._
- [x] **searchLocale**: _Locale to use in searching. The default locale is the host environment's current locale._
- [x] **searchMessage**: _Text to be displayed in hidden accessible field when filtering returns any results. Defaults to value from PrimeVue locale configuration._
- [x] **selectionMessage**: _Text to be displayed in hidden accessible field when options are selected. Defaults to value from PrimeVue locale configuration._
- [x] **emptySelectionMessage**: _Text to be displayed in hidden accessible field when any option is not selected. Defaults to value from PrimeVue locale configuration_
- [x] **emptySearchMessage**: _Text to display when filtering does not return any results. Defaults to value from PrimeVue locale configuration._
- [x] **tabindex**: _Index of the element in tabbing order._
- [x] **ariaLabel**: _Defines a string value that labels an interactive element._
- [x] **ariaLabelledby**: _Identifier of the underlying input element._

**_Deprecated Properties:_**

- [x] **field**: _since v3.16.0. Use 'optionLabel' property instead_
